randovania/layout/lib/teleporters.py

```diff
@@ -82,8 +82,9 @@ def _valid_teleporter_target(area: Area, game: RandovaniaGame):
 
 class TeleporterTargetList(location_list.LocationList):
     @classmethod
-    def areas_list(cls, game: RandovaniaGame):
-        return location_list.area_locations_with_filter(game, lambda area: _valid_teleporter_target(area, game))
+    def nodes_list(cls, game: RandovaniaGame):
+        temp_area_list = location_list.area_with_filter(game, lambda area: _valid_teleporter_target(area, game))
+        return location_list.node_locations_with_filter(game, lambda node: node.identifier.area_identifier in temp_area_list and temp_area_list[node.identifier.area_identifier].default_node == node.name)
```

Comment:
This looks too complicated.
Check if there is another method to apply the `_valid_teleporter_target` logic to an area and only yield the `default_node` from it.